### PR TITLE
Add support for tasks and unordered list markdown

### LINKED ISSUE #4927: Add support for tasks and unordered list markdown
### Description:
Tasks and unordered list does not appear in correct format as that in RC web app.

### Environment Information:

- Rocket.Chat Server Version: open.rocket.chat (6.0.0)

### Steps to reproduce:

1. Tasks should display in expected form
2. Unordered list should display in expected form 

### Expected behavior:
image

### Actual behavior:
WhatsApp Image 2023-02-27 at 17 22 36

### PR DESCRIPTION
**Pull request by imf-ali**
## Issue(s)	
Tasks and unordered list does not appear in correct format as that in RC web app.

## How to test or reproduce
- [x] Tasks should display in expected form
- [x] Unordered list should display in expected form

## Screenshots
Previous behaviour
image

Improvement
image

## Types of changes

- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] Improvement (non-breaking change which improves a current function)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Documentation update (if none of the other choices apply)

## Checklist

- [x] I have read the [CONTRIBUTING](https://github.com/RocketChat/Rocket.Chat/blob/develop/.github/CONTRIBUTING.md#contributing-to-rocketchat) doc
- [x] I have signed the [CLA](https://cla-assistant.io/RocketChat/Rocket.Chat.ReactNative)
- [x] Lint and unit tests pass locally with my changes
- [x] I have added tests that prove my fix is effective or that my feature works (if applicable)
- [ ] I have added necessary documentation (if applicable)
- [ ] Any dependent changes have been merged and published in downstream modules

## Further comments

Fix #4927

### CODE REVIEW

**Comment by imf-ali** [comment]
Hey @diegolmello and @GleidsonDaniel , please have a look at the PR.

File: app/containers/markdown/new/TaskList.tsx [hunk 1]
```diff
@@ -16,7 +17,9 @@ const TaskList = ({ value = [] }: ITasksProps) => {
 		<View>
 			{value.map(item => (
 				<View style={styles.row}>
-					<Text style={[styles.text, { color: colors.bodyText }]}>{item.status ? '- [x] ' : '- [ ] '}</Text>
+					<Text style={[styles.text, { color: colors.bodyText }]}>
+						<CheckBox disabled={true} style={{ height: 20, width: 20, margin: 5 }} boxType='square' value={item.status} />
```
[the remark this answers is not in the document]
Comment (imf-ali) [reply app/containers/markdown/new/TaskList.tsx]: Done.

**Comment by imf-ali** [comment]
@GleidsonDaniel please have a look at it.

**Comment by GleidsonDaniel** [comment]
> A quick comment, instead add a new library to the repo, can you use it the same way as other places?
> 
> As you can see here:
> 
> https://github.com/RocketChat/Rocket.Chat.ReactNative/blob/2d07b1682ca8ca06eb1addfc8b6bbd3287f220ae/app/containers/MessageBox/index.tsx#L1119-L1124
> 
> we are using the custom icon for the checkbox

@imf-ali i think this is the best approach.
Change your PR following this pattern commented by @reinaldonetof

**Review by GleidsonDaniel — CHANGES_REQUESTED** [review]
Change your PR following the pattern commented by @reinaldonetof

**Comment by imf-ali** [comment]
Changes done. Please have a look @GleidsonDaniel @reinaldonetof .
Preview of the change
image

File: ios/Podfile.lock [hunk 2]
```diff
@@ -505,6 +506,9 @@ PODS:
     - React-Core
   - RNCAsyncStorage (1.17.11):
     - React-Core
+  - RNCCheckbox (0.5.15):
```
Comment (GleidsonDaniel) [inline ios/Podfile.lock:509]: update your podfile

**Review by GleidsonDaniel — CHANGES_REQUESTED** [review]

File: ios/Podfile.lock (hunk 2 above)
Comment (imf-ali) [reply ios/Podfile.lock]: Done

**Comment by imf-ali** [comment]
@GleidsonDaniel comment addressed. Please have a look.

File: ios/Podfile.lock [hunk 3]
```diff
@@ -997,4 +997,4 @@ SPEC CHECKSUMS:
 
 PODFILE CHECKSUM: da36ddb728f8698fcf3e8b17d853ccedc4f8a24c
 
-COCOAPODS: 1.12.1
+COCOAPODS: 1.11.3
```
Comment (GleidsonDaniel) [inline ios/Podfile.lock:1000]:
```suggestion
COCOAPODS: 1.12.1
```

**Review by GleidsonDaniel — CHANGES_REQUESTED** [review]

File: ios/Podfile.lock (hunk 3 above)
Comment (imf-ali) [reply ios/Podfile.lock]: Done.

File: app/containers/markdown/new/TaskList.tsx [hunk 4]
```diff
@@ -16,7 +17,14 @@ const TaskList = ({ value = [] }: ITasksProps) => {
 		<View>
 			{value.map(item => (
 				<View style={styles.row}>
-					<Text style={[styles.text, { color: colors.bodyText }]}>{item.status ? '- [x] ' : '- [ ] '}</Text>
+					<Text style={[styles.text, { color: colors.bodyText }]}>
+						<CustomIcon
+							testID={item.status ? 'send-to-channel-checked' : 'send-to-channel-unchecked'}
+							name={item.status ? 'checkbox-checked' : 'checkbox-unchecked'}
+							size={24}
+							color='#9297a2'
```
Comment (GleidsonDaniel) [inline app/containers/markdown/new/TaskList.tsx:25]: Don't put the color directly in the code, use a constant, preferably one that is in the app's const colors, as it doesn't change color, add it as a fixed color.

File: app/containers/markdown/new/TaskList.tsx [hunk 5]
```diff
@@ -16,7 +17,14 @@ const TaskList = ({ value = [] }: ITasksProps) => {
 		<View>
 			{value.map(item => (
 				<View style={styles.row}>
-					<Text style={[styles.text, { color: colors.bodyText }]}>{item.status ? '- [x] ' : '- [ ] '}</Text>
+					<Text style={[styles.text, { color: colors.bodyText }]}>
+						<CustomIcon
+							testID={item.status ? 'send-to-channel-checked' : 'send-to-channel-unchecked'}
```
Comment (GleidsonDaniel) [inline app/containers/markdown/new/TaskList.tsx:22]:
```suggestion
							testID={item.status ? 'task-list-checked' : 'task-list-unchecked'}
```

**Review by GleidsonDaniel — CHANGES_REQUESTED** [review]

**Comment by imf-ali** [comment]
> I saw that you sent me a private message asking for the review. Just pay attention to one issue: Previously your PR was not reviewed because it didn't even pass the lint, so before requesting the PR check that all the steps in the checklist are correct.

Oh, sorry for the inconvenience. Will make sure next time, all the checks are successful. I have addressed the remaining comments. Kindly take a look once.

**Review by GleidsonDaniel — APPROVED** [review]

**Pull request merged**